src/Transformation/Transformation.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Transformation;
+
+/**
+ * A transformation is a function from one datatype to another.
+ *
+ * It MUST NOT perform any sideeffects, i.e. it must be morally impossible to observe
+ * how often the transformation was actually performed. It MUST NOT touch the providedi
+ * value, i.e. it is allowed to create new values but not to modify existing values.i
+ * This would be an observable sideeffect.  
```

Comment:
I fail to see how this could get outdated. It is not some kind of process rule or general agreement but the exact thing the abstraction should encode. I would reference the doc of the interface from the readme, if we would want to get rid of this.